### PR TITLE
typescript & yarn monorepo

### PR DESCRIPTION
**Pull request by ermalkaleci**

### CODE REVIEW

**Review by shunjizhan — APPROVED** [review]
LGTM, just a couple small comments and questions. 

Nice to see rush gone lol, and now the install & build process is pretty smooth!

Let's not worry about the prettier stuff, I can probably import some eslint rules to fight it back later

File: .gitattributes [hunk 1]
```diff
@@ -4,11 +4,3 @@ pnpm-lock.yaml               merge=binary linguist-generated=true
 shrinkwrap.yaml              merge=binary linguist-generated=true
 npm-shrinkwrap.json          merge=binary linguist-generated=true
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply .gitattributes:5]: Yes there are still some garbage

File: packages/eth-providers/src/base-provider.ts [hunk 2]
```diff
@@ -1076,8 +1072,10 @@ export abstract class BaseProvider extends AbstractProvider {
    * @param transaction The transaction to estimate the resources of
    * @returns The estimated resources used by this transaction
    */
-  estimateResources = async (transaction: Deferrable<TransactionRequest>): Promise<{
-    usedGas: BigNumber,
+  estimateResources = async (
```
[the remark this answers is not in the document]
Comment (shunjizhan) [reply packages/eth-providers/src/base-provider.ts:1075]: I guess it's incentive is "break longer lines, and stretch shorter lines", in general it has to do something to show ppl it "prettier" the code

**Comment by ermalkaleci** [comment]
> LGTM, just a couple small comments and questions. 
> 
> 
> 
> Nice to see rush gone lol, and now the install & build process is pretty smooth!
> 
> 
> 
> Let's not worry about the prettier stuff, I can probably import some eslint rules to fight it back later

I don't care about prettier, eslint is fine

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -20,18 +20,37 @@ jobs:
...
-        run: node common/scripts/install-run-rush.js build --verbose --to @acala-network/bodhi --to @acala-network/eth-rpc-adapter
+        run: yarn build
 
-      - name: publish
-        run: node common/scripts/install-run-rush.js publish --apply --publish --include-all --target-branch master --add-commit-details --set-access-level public -n ${{ secrets.NPM_AUTH_TOKEN }}
+      - run: |
+          echo npmAuthToken: "\${NPM_AUTH_TOKEN}" >> ./.yarnrc.yml
+
+      - name: publish @acala-network/eth-transactions
+        run: yarn workspace @acala-network/eth-transactions npm publish --tolerate-republish --access public
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply .github/workflows/publish.yml:39]: Sure we can do that

File: examples/waffle/incentives/test/incentives.test.ts [hunk 4]
```diff
@@ -172,5 +173,7 @@ describe('incentives', () => {
     await expect(incentivesPredeployed.claimRewards(1, ADDRESS.LP_ACA_AUSD))
       .to.emit(incentivesPredeployed, 'ClaimedRewards')
       .withArgs(await wallet.getAddress(), 1, ADDRESS.LP_ACA_AUSD);
+
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply examples/waffle/incentives/test/incentives.test.ts:177]: Left over

File: packages/eth-providers/src/base-provider.ts [hunk 5]
```diff
@@ -342,11 +346,9 @@ export abstract class BaseProvider extends AbstractProvider {
   }
 
   startSubscriptions = async (): Promise<void> => {
-    await Promise.all([
-      this._subscribeEventListeners(),
-      this._subscribeNewBlocks(),
-      this._subscribeRuntimeVersion(),
-    ].flat());
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply packages/eth-providers/src/base-provider.ts:349]: We will need to change those back manually

File: vitest.config.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { defineConfig } from 'vitest/config';
+import swc from 'unplugin-swc';
+import tsconfigPaths from 'vite-tsconfig-paths';
+
+export default defineConfig({
+  test: {
+    testTimeout: 300000,
+    hookTimeout: 60000,
+    dangerouslyIgnoreUnhandledErrors: true,
```
[the remark this answers is not in the document]
Comment (ermalkaleci) [reply vitest.config.ts:9]: Yes better to remove

**Comment by shunjizhan** [comment]
yeah let's keep it as what it is now, as the changes are already made, doesn't really matter

**Comment by shunjizhan** [comment]
I like how docker mounts local volumes BTW, it's much faster

File: packages/eth-providers/src/base-provider.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (shunjizhan) [reply packages/eth-providers/src/base-provider.ts:349]: it's funny that I found this example in prettier's website intro, however prettier is doing exactly opposite thing

Screenshot 2023-04-18 at 16 46 11

**Pull request merged**